### PR TITLE
mod : No repair below 12 players + message if low population in rewards

### PR DESCRIPTION
**Pull request by namidaka**

### CODE REVIEW

File: src/Module.Server/Rewards/CrpgRewardClient.cs [hunk 1]
```diff
@@ -43,6 +43,12 @@ protected override void AddRemoveMessageHandlers(GameNetwork.NetworkMessageHandl
     private void HandleRewardUser(CrpgRewardUser message)
     {
         var reward = message.Reward;
+        if (message.LowPopulation)
+        {
+            InformationManager.DisplayMessage(new InformationMessage("Low Population! Repair Are Free!",
+                new Color(65, 105, 225)));
+        }
+
```
Comment (verdie-g) [inline src/Module.Server/Rewards/CrpgRewardClient.cs:51]: What do you think of merging that into the gold reward message. For example
```
Gained 45 gold (reward: 45, upkeep: free, low population server)
```

File: src/Module.Server/Rewards/CrpgRewardServer.cs [hunk 2]
```diff
@@ -243,7 +243,7 @@ public override void OnRemoveBehavior()
             if (missionPeer.Team != null && missionPeer.Team.Side != BattleSideEnum.None)
             {
                 int crpgUserId = crpgPeer.User.Id;
-                var crpgUserDamagedItems = BreakItems(crpgPeer, durationRewarded * (isLowPopulationServer ? 0.2f : 1f));
+                var crpgUserDamagedItems = BreakItems(crpgPeer, durationRewarded * (isLowPopulationServer ? 0f : 1f));
```
Comment (verdie-g) [inline src/Module.Server/Rewards/CrpgRewardServer.cs:246]: You can avoid the call to GetBrokenItemsByCrpgUserId altogether when it's low pop

File: src/Module.Server/Rewards/CrpgRewardServer.cs [hunk 3]
```diff
@@ -135,15 +135,15 @@ public override void OnRemoveBehavior()
             return;
         }
 
-        bool lowPopulationServer = networkPeers.Length < 4;
-
+        bool verylowPopulationServer = networkPeers.Length < 2;
```
Comment (verdie-g) [inline src/Module.Server/Rewards/CrpgRewardServer.cs:138]: camelCase -> veryLowPopulationServer

**Review by verdie-g — CHANGES_REQUESTED** [review]

File: src/Module.Server/Rewards/CrpgRewardClient.cs (hunk 1 above)
Comment (namidaka) [reply src/Module.Server/Rewards/CrpgRewardClient.cs:51]: That's a really nice idea

Comment (namidaka) [reply src/Module.Server/Rewards/CrpgRewardClient.cs]: it's done

File: src/Module.Server/Rewards/CrpgRewardServer.cs (hunk 2 above)
Comment (namidaka) [reply src/Module.Server/Rewards/CrpgRewardServer.cs]: it's done

File: src/Module.Server/Rewards/CrpgRewardServer.cs (hunk 3 above)
Comment (namidaka) [reply src/Module.Server/Rewards/CrpgRewardServer.cs]: it's done

File: src/Module.Server/Rewards/CrpgRewardClient.cs [hunk 4]
```diff
@@ -61,9 +56,11 @@ private void HandleRewardUser(CrpgRewardUser message)
             (Color color, string verb) = gain > 0
                 ? (new Color(65, 105, 225), "Gained")
                 : (new Color(0.74f, 0.28f, 0.01f), "Lost");
+            string rewardMessage = message.LowPopulation
+                ? $"{verb} {gain} gold (reward: {reward.Gold}, upkeep: free , low population server)."
+                : $"{verb} {gain} gold (reward: {reward.Gold}, upkeep: {message.RepairCost}).";
             InformationManager.DisplayMessage(
-                new InformationMessage($"{verb} {gain} gold (reward: {reward.Gold}, upkeep: {message.RepairCost}).",
-                    color));
+                new InformationMessage(rewardMessage, color));
```
Comment (verdie-g) [inline src/Module.Server/Rewards/CrpgRewardClient.cs:63]: What about
```csharp
            string upkeepStr = message.LowPopulation
                ? "free , low population server"
                : message.RepairCost.ToString()
            InformationManager.DisplayMessage(
                new InformationMessage($"{verb} {gain} gold (reward: {reward.Gold}, upkeep: {upkeepStr}).",
                color));

```

**Review by verdie-g — CHANGES_REQUESTED** [review]

File: src/Module.Server/Rewards/CrpgRewardClient.cs [hunk 5]
```diff
@@ -55,8 +56,12 @@ private void HandleRewardUser(CrpgRewardUser message)
             (Color color, string verb) = gain > 0
                 ? (new Color(65, 105, 225), "Gained")
                 : (new Color(0.74f, 0.28f, 0.01f), "Lost");
+            string upkeepMessage = message.LowPopulation
+                ? "free , low population server"
```
Comment (verdie-g) [inline src/Module.Server/Rewards/CrpgRewardClient.cs:60]: There is no space before commas but one after -> "free, low..."

**Review by verdie-g — CHANGES_REQUESTED** [review]

File: src/Module.Server/Rewards/CrpgRewardClient.cs (hunk 5 above)
Comment (namidaka) [reply src/Module.Server/Rewards/CrpgRewardClient.cs]: done

**Review by verdie-g — APPROVED** [review]

**Pull request merged**